File: Algorithm_Cpp/Algorithm_Cpp/Deque/deque1.cpp
#include <iostream>
#include <string>
#include <deque>

using namespace std;


int deque1() {

        deque<int> dq;
        
        int x;
        cin >> x;
        string s;
        
        for(int i = 0; i < x; i++){
            cin >> s;
            if(s == "push_front"){
                int y;
                cin >> y;
                dq.push_front(y);
            }
            else if(s == "push_back"){
                int y;
                cin >> y;
                dq.push_back(y);
            }
            else if(s == "pop_front"){
                cout << dq.front() << endl;;
                dq.pop_front();
            }
            else if(s == "pop_back"){
                cout << dq.back() << endl;;
                dq.pop_back();
            }
            else if(s == "size"){
                cout << dq.size() << endl;;
            }
            else if(s == "empty"){
                if(dq.empty()){
                    cout << "1" << endl;
                }
                else cout << "0" << endl;;
            }
            else if(s == "front"){
                cout << dq.front() << endl;;
            }
            else if(s == "back"){
                cout << dq.back() << endl;;
            }
        }
    return 0;
}

File: Algorithm_Java/src/Deque/deque1.java
package Deque;

import java.util.Deque;
import java.util.ArrayDeque;
import java.util.Scanner;

public class deque1 {
    public static void main(String[] args) {

        Deque<Integer> dq = new ArrayDeque<>();

        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();

        for(int i = 0; i < x; i++){
            String s = sc.next();
            if(s.equals("push_front")){
                int y = sc.nextInt();
                dq.addFirst(y);
            }
            else if(s.equals("push_back")){
                int y = sc.nextInt();
                dq.addLast(y);
            }
            else if(s.equals("pop_front")){
                System.out.println(dq.peekFirst());
                dq.pollFirst();
            }
            else if(s.equals("pop_back")){
                System.out.println(dq.peekLast());
                dq.pollLast();
            }
            else if(s.equals("size")){
                System.out.println(dq.size());
            }
            else if(s.equals("empty")){
                if(dq.isEmpty()){
                    System.out.println("1");
                }
                else System.out.println("0");
            }
            else if(s.equals("front")){
                System.out.println(dq.peekFirst());
                dq.peekFirst();
            }
            else if(s.equals("back")){
                System.out.println(dq.peekLast());
                dq.peekLast();
            }
        }



    }
}
